saleor/graphql/webhook/mutations.py

```diff
@@ -419,3 +425,115 @@ def perform_mutation(cls, _root, info, **data):
                 event_type, object, query, request
             )
         return WebhookDryRun(payload=payload)
+
+
+class WebhookTrigger(BaseMutation):
+    delivery = graphene.Field(EventDelivery)
+
+    class Arguments:
+        webhook_id = graphene.ID(description="The ID of the webhook.", required=True)
+        object_id = graphene.ID(
+            description="The ID of an object to serialize.", required=True
+        )
+
+    class Meta:
+        description = (
+            "Trigger a webhook event. Supports a single event (the first, if multiple "
+            "provided in the `webhook.subscription_query`). Requires permission "
+            "relevant to processed event. Successfully delivered webhook does not "
+            "return `delivery`." + ADDED_IN_311 + PREVIEW_FEATURE
+        )
+        permissions = (AuthorizationFilters.AUTHENTICATED_STAFF_USER,)
+        error_type_class = WebhookTriggerError
+
+    @classmethod
+    def validate_input(cls, info, **data):
```

Comment:
IMO we can split this validation method into smaller ones to increase readability.